js/web/lib/wasm/jsep/webgpu/ops/reduce.ts

```diff
@@ -97,23 +102,34 @@ const createReduceProgramInfo =
       };
     };
 
-const createReduceAttributesFromInput = (input: TensorView, attributes: ReduceAttributes): ReduceAttributes => {
-  const axes: number[] = [];
-  input.getBigInt64Array().forEach(v => axes.push(Number(v)));
-  const keepDims = attributes.keepDims;
-  const noopWithEmptyAxes = attributes.noopWithEmptyAxes;
-  return createAttributeWithCacheKey({axes, keepDims, noopWithEmptyAxes});
-};
+const createReduceAttributesFromInputs =
+    (inputs: readonly TensorView[], attributes: ReduceAttributes): ReduceAttributes => {
+      const axes: number[] = [];
+      // Handle empty axes, zero-sized input and zero input.
+      const sum = inputs[1].dims.reduce((sum, v) => sum + v, 0);
+      const prod = inputs[1].dims.reduce((prod, v) => prod * v, 1);
```

Comment:
`inputs[1].dims.length` should always be 1, so `sum` = `prod` = `inputs[1].dims[0]`

so axes can be simplified to 
`const axes = inputs[1]?.dims[0] > 0 ? Array.from(inputs[1].getBigInt64Array(), Number) : attributes.axes;`

Reply:
actually - if inputs[1] exist, we should always ignore attributes.axes, right? so it should be 
`const axes = inputs[1]?.dims[0] > 0 ? Array.from(inputs[1].getBigInt64Array(), Number) : [];`

Reply:
and if `inputs[1]?.dims[0] === 0`, `Array.from(inputs[1].getBigInt64Array(), Number)` is an empty array. so we can finally simply it to `const axes = Array.from(inputs[1].getBigInt64Array(), Number);`

Reply:
If input[1] is not a vector and if is simply 0 then inputs[1].getBigInt64Array() is causing strange error. According to specification input[1] should a one dim array but some tests cases have it just 0. I want to make sure that it doesn't cause error by checking both sum and prod.